prov/gni/test/allocator.c

```diff
@@ -151,17 +151,22 @@ static int verify_hugepages(void)
 
 /*
  * Open an allocator with the given parameters and immediately close it. Verify
- * that everything returned a successful error code.
+ * that everything returned a successful error code.  Note that for large
+ * page sizes over ~64 MB, it can be iffy whether or not large pages can
+ * be synthesized if the linux page cache has become highly fragmented, so
+ * we have a may fail parameter that checks to see if the error return is
+ * -FI_ENOMEM, in which case don't treat as fatal.
  */
 static void open_close_allocator(enum gnix_page_size page_size,
 				 size_t mbox_size,
-				 size_t mpmmap)
+				 size_t mpmmap, bool may_fail)
 {
 	int ret;
 
 	ret = _gnix_mbox_allocator_create(ep_priv->nic, NULL, page_size,
 					  mbox_size, mpmmap, &allocator);
-	cr_assert_eq(ret, FI_SUCCESS, "_gnix_mbox_allocator_create failed.");
+	if ((ret == -FI_ENOMEM) && (may_fail == true)) return;
```

Comment:
Can you print out a warning that the test was skipped when this happens?  If we see it 100% of the time, it's worth looking for a problem inside _gnix_mbox_allocator_create.